### PR TITLE
build: fix check_set_compiler_property

### PR DESCRIPTION
**Pull request by yperess**
check_set_compiler_property() was calling zephyr_check_compiler_flag without first having set 'check'. This is required since https://github.com/zephyrproject-rtos/zephyr/commit/c32e4a71bf867029b6e828e1c1e49865c48b1c52 added a second dereferencing for the 'check' argument.

Issue #53124

### CODE REVIEW

**Review by tejlmand — CHANGES_REQUESTED** [review]
commit message refers to this commit: https://github.com/zephyrproject-rtos/zephyr/commit/c32e4a71bf867029b6e828e1c1e49865c48b1c52

but that commit doesn't match your commit message, did you perhaps meant to refer to:
https://github.com/zephyrproject-rtos/zephyr/commit/04a27651ea7402ecd24fa907983416ed3fc26a12 ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
not convinced this is a proper fix.
See also: https://github.com/zephyrproject-rtos/zephyr/issues/53124#issuecomment-1455920875

File: cmake/modules/extensions.cmake [hunk 1]
```diff
@@ -2116,6 +2116,7 @@ function(check_set_compiler_property)
       separate_arguments(option UNIX_COMMAND ${option})
     endif()
 
+    set(check)
```
[the remark this answers is not in the document]
Comment (yperess) [reply cmake/modules/extensions.cmake:2119]: I didn't make any custom calls to `check_set_compiler_property` on my end. I added a second commit to this PR so you can see how I've tested this. I printed the `option` then `FOUND` for both `CXX` and `C` if the check passes. The relevant logs are:

```
option='-Wno-pointer-sign'
    CXX: FOUND
    C: FOUND
option='-Werror=implicit-int'
    CXX: FOUND
    C: FOUND
```

[the remark this answers is not in the document]
Comment (tejlmand) [reply cmake/modules/extensions.cmake]: using this PR and building the `cpp_synchronization` using Zephyr SDK it prints the following:
```
$ git rev-parse HEAD
a86c5350ff7ce22cf08f426174d10ce71bd25e50
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild samples/cpp/cpp_synchronization/
Loading Zephyr default modules (Zephyr repository).
-- Application: /projects/github/ncs/zephyr/samples/cpp/cpp_synchronization
...
option='-Wno-main'
    CXX: FOUND
    C: FOUND
option='-Wno-pointer-sign'
    C: FOUND
option='-Wpointer-arith'
    CXX: FOUND
    C: FOUND
...
option='-Wno-unused-but-set-variable'
    CXX: FOUND
    C: FOUND
option='-Werror=implicit-int'
    C: FOUND
option='-mstack-protector-guard=global'
    CXX: FOUND
    C: FOUND
...
```

and exactly as expected, those flags that are present in `CXX_EXCLUDED_OPTIONS` returns a `0` (false) in the check variable and thus your debug code doesn't rpint `CXX: FOUND` in those cases.

Could you please verify that this is indeed a problem in Zephyr, cause so far I suspect that you might have local patches in the fork where you see an error.

**Review by tejlmand — CHANGES_REQUESTED** [review]
current investigations have shown no errors in Zephyr.

**Comment by tejlmand** [comment]
@yperess thanks for the description in #55490, was able to reproduce.
Proper fix posted here: #55516

**Pull request closed without merging**